### PR TITLE
Show region constraints on workspace (WOR-1243).

### PR DESCRIPTION
**Pull request by cahrens**
### Jira Ticket: https://broadworkbench.atlassian.net/browse/WOR-1243

For workspaces with a region constraint set (currently, can only be Azure workspaces that are somewhat "hand-crafted"), displays the regions that resources must be in.

image

image

Example with both protected and region constraint:

image

### CODE REVIEW

File: src/libs/azure-regions.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// This is in its own file to avoid circular dependency issues.
```
Comment (cahrens) [inline src/libs/azure-regions.ts:1]: I tried splitting up `workspace-utils` in hopes of solving the circular dependency issue (which came out of `workspace-utils` relying on `azure-utils`, which in turn relies on `runtime-utils`, which reaches back into `workspace-utils`), but I was having trouble finding something that made semantic sense and solved the problem… and it involved changing 50 some files. So I decided to simply move this data structure that `workspace-utils` wants access to into its own file.

File: src/libs/azure-regions.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
+// This is in its own file instead of azure-utils to avoid circular dependency issues.
```
Comment (cahrens) [inline src/libs/azure-regions.ts:1]: I tried splitting up `workspace-utils` in hopes of solving the circular dependency issue (which came out of `workspace-utils` relying on `azure-utils`, which in turn relies on `runtime-utils`, which reaches back into `workspace-utils`), but I was having trouble finding something that made semantic sense and solved the problem… and it involved changing 50 some files. So I decided to simply move this data structure that `workspace-utils` wants access to into its own file.

File: src/pages/workspaces/workspace/WorkspaceAttributeNotice.ts [hunk 3]
```diff
@@ -48,7 +49,7 @@ const Notice = (props: NoticeProperties) => {
       h(
         TooltipTrigger,
         {
-          content: [div({ key: props.label }, [props.tooltip])],
+          content: [div({ key: props.label, style: { maxWidth: 300 } }, [props.tooltip])],
```
Comment (cahrens) [inline src/pages/workspaces/workspace/WorkspaceAttributeNotice.ts:52]: Lou asked that I find a width that minimized "widows"… which I think means the tooltip text breaks in non-optimal places. This max width seemed to work pretty well for our known tooltip text.

File: src/libs/workspace-utils.ts [hunk 4]
```diff
@@ -108,6 +109,34 @@ export const containsProtectedDataPolicy = (policies: WorkspacePolicy[] | undefi
...
+    (policy) => policy.namespace === 'terra' && policy.name === 'region-constraint',
+    policies
+  );
+  const regionLabels: string[] = [];
+  _.forEach((policy) => {
+    _.forEach((data) => {
+      if ('region-name' in data) {
+        const region = data['region-name'];
+        const regionName = region.startsWith('azure.') ? region.split('azure.')[1] : region;
+        regionLabels.push(_.has(regionName, azureRegions) ? azureRegions[regionName].label : regionName);
```
Comment (cahrens) [inline src/libs/workspace-utils.ts:129]: I decided that if we can't parse the region into something we recognize, it's better to show them what the policy actually specifies vs. a generic "unknown" value.

**Review by marctalbott — APPROVED** [review]

File: src/libs/azure-regions.ts (hunk 2 above)
Comment (blakery) [reply src/libs/azure-regions.ts:1]: You made a valiant attempt.
TBH, I think the separate file is better anyway.

**Review by blakery — APPROVED** [review]

**Pull request merged**